src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -92,7 +92,7 @@ private bool ResetSolutionSettingsIfNeeded()
 
             if (_solutionSettings == null)
             {
-                _solutionSettings = new Tuple<string, Lazy<ISettings>>(null, new Lazy<ISettings>(() => NullSettings.Instance));
```

Comment:
Looking at the history in 4.9, this was being set to NullSettings.Instance 

https://github.com/NuGet/NuGet.Client/blob/release-4.9.4-rtm/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs#L80-L83

Can you explain why it manifests as a regression?

Reply:
The change was to the section above: https://github.com/NuGet/NuGet.Client/commit/ec48ee474965d3d5f41ed7e5b4199af79868351a#diff-6545c160a4f64189999990c5a6961eb6L73

Before, it would always read the settings, only if there was an exception it would use the null provider. After the change, it does a comparison `if (!string.Equals(root, _solutionSettings?.Item1))`, but on startup both `root` and `_solutionSettings` are null, so they're equal, so the `if` condition is false and there is no initial load.

On one hand a configuration error is still going to be a problem, so we should use the null provider in that case. On the other hand, since it's lazy loaded now, the exception handling we have will never be used. In any case I now realise that I need to change my PR to improve this.

Reply:
better fix available now.